consensus/poet/simulator/sawtooth_poet_simulator/poet_enclave_simulator/poet_enclave_simulator.py

```diff
@@ -343,78 +339,72 @@ def create_wait_timer(cls,
             wait_timer.signature = \
                 signing.sign(
                     wait_timer.serialize(),
-                    cls._poet_private_key)
-
-            # Keep track of the active wait timer
-            cls._active_wait_timer = wait_timer
+                    poet_private_key)
 
             return wait_timer
 
     @classmethod
     def deserialize_wait_timer(cls, serialized_timer, signature):
-        with cls._lock:
-            # Verify the signature before trying to deserialize
-            if not signing.verify(
-                    serialized_timer,
-                    signature,
-                    cls._poet_public_key):
-                return None
-
         return \
             EnclaveWaitTimer.wait_timer_from_serialized(
                 serialized_timer=serialized_timer,
                 signature=signature)
 
     @classmethod
     def create_wait_certificate(cls,
+                                sealed_signup_data,
                                 wait_timer,
                                 block_hash):
         with cls._lock:
-            # If we don't have a PoET private key, then the enclave has not
-            # been properly initialized (either by calling create_signup_info
-            # or unseal_signup_data)
-            if cls._poet_private_key is None:
+            # Extract keys from the 'sealed' signup data
+            if sealed_signup_data is None:
+                raise ValueError('Sealed Signup Data is None')
+            signup_data = \
+                json2dict(
+                    base64.b64decode(sealed_signup_data.encode()).decode())
+            poet_private_key = signup_data['poet_private_key']
+            poet_public_key = signup_data['poet_public_key']
+
+            if poet_private_key is None or poet_public_key is None:
                 raise \
                     ValueError(
-                        'Enclave must be initialized before attempting to '
-                        'create a wait certificate')
+                        'Invalid signup data. No poet key(s).')
 
             # Several criteria need to be met before we can create a wait
             # certificate:
-            # 1. We have an active timer
-            # 2. The caller's wait timer is the active wait timer.  We are not
-            #    going to rely the objects, being the same, but will compute
-            #    a signature over the object and verify that the signatures
-            #    are the same.
-            # 3. The active timer has expired
-            # 4. The active timer has not timed out
+            # 1. This signup data was used to sign this timer.
```

Comment:
The previous code used a private variable to track an "active wait timer". I took that out (like removing poet_state). The updated comment was meant to remove the allusion to that old private variable. This function still requires a wait timer to be passed in. The concept of whether or not it is "active" is spread across checks like signature validity and time.